illumina.py

```diff
@@ -400,44 +401,52 @@ def main_common_barcodes(args):
 
 __commands__.append(('common_barcodes', parser_common_barcodes))
 
-def count_and_sort_barcodes(barcodes_dir, outSummary, truncateToLength=None, includeNoise=False, omitHeader=False):
+def count_and_sort_barcodes(barcodes_dir, outSummary, truncateToLength=None, includeNoise=False, omitHeader=False, threads=None):
     # collect the barcode file paths for all tiles
     tile_barcode_files = [os.path.join(barcodes_dir, filename) for filename in os.listdir(barcodes_dir)]
 
     # count all of the barcodes present in the tile files
     log.info("reading barcodes in all tile files")
-    barcode_counts = defaultdict(lambda: 0)
 
-    def sum_reducer(accumulator, element):
-        for key, value in element.items():
-            accumulator[key] = accumulator.get(key, 0) + value
-        return accumulator
+    reduce_db_file    = util.file.mkstempfname('sqlite_.db')
+    reduce_db     = util.file.CountDB(reduce_db_file)
+    reduce_db.start()
```

Comment:
with some proposed changes in util.file, can this whole thing become a `with util.file.CountDB() as reduce_db:`? I forget how python inheritance works, but I think when DBConnection.__init__ calls self.start() at the end, it should be calling the CountDB.start method...

Reply:
Yep, it exposes `__enter__()`, `__exit()`, and `__init__()` so it can be a context manager. I'll add the temp cleanup to `close()` for the `db_file=None` case.